utils/R/convert.input.R

```diff
@@ -138,57 +217,100 @@ convert.input <- function(input.id, outfolder, formatname, mimetype, site.id, st
       result$mimetype[i] <- mimetype
       result$formatname[i] <- formatname    
     }
-  
-  } else if (conversion == "local.remote") { # perform conversion on local or remote host
-     if (missing(format.vars)) {
-       args = c(dbfile$file_path, dbfile$file_name,outfolder,start_date,end_date) 
-       if(!is.null(names(l))){
-         cmdFcn = paste0(
-           paste0(pkg, "::", fcn, "(", paste0("'", args, "'", collapse=",")), ",",
-           paste( paste(names(l), "=", paste0("'", unlist(l), "'")), collapse=","), ")")
-       }else{
-         cmdFcn  = paste0(pkg,"::",fcn,"(",paste0("'",args,"'",collapse=","),")") 
-       } 
-    } else {
-       args = c(dbfile$file_path,dbfile$file_name,outfolder,start_date,end_date) 
-       cmdFcn  = paste0(pkg,"::",fcn,"(",paste0("'",args,"'",collapse=","),",format=",paste0(list(format.vars)),")")
-    } 
-    print(cmdFcn) #do we want to print this?
+  } else if (conversion == "local.remote") { 
+    # perform conversion on local or remote host
+
+    fcn.args <- input.args
+    fcn.args$overwrite <- overwrite
+    fcn.args$in.path <- dbfile$file_path
+    fcn.args$in.prefix <- dbfile$file_name
+    fcn.args$outfolder <- outfolder
+    fcn.args$start_date <- start_date
+    fcn.args$end_date <- end_date 
+
+    arg.string <- listToArgString(fcn.args)
```

Comment:
Here's the new approach to handling argument passing. Start with the `...` args (called input.args here), which are already a named list. Then add any other args that you know you want, with the desired names. Then pass them to `listToArgString` to get a string version you can paste into the remote call. This is pretty flexible, and importantly since all arguments are named it lets you omit them willy-nilly as long as whatever fcn you're calling accepts `...` (so it doesn't bonk out when receiving arguments it can't use) and you don't omit anything that it actually requires. 
So for example, the download.* functions still receive `in.path` and `in.prefix` args (which will be set to NULL since no dbfile will have been found), but they pass through `...` unnoticed since they're not used.